exercise.wwwapi/Repositories/ProductRepository.cs

```diff
@@ -0,0 +1,82 @@
+﻿using exercise.wwwapi.Data;
+using exercise.wwwapi.Models;
+using exercise.wwwapi.Repository.Interfaces;
+
+namespace exercise.wwwapi.Repository
+{
+    class ProductRepository : IRepository
+    {
+        private ProductContext _db;
+
+        public ProductRepository(ProductContext db)
+        {
+            _db = db;
+        }
+
+        public ProductItem? Add(ProductPayload payload)
+        {
+            var product = new ProductItem() { Name = payload.Name, Price = payload.Price, Category = payload.Category };
+
+            if (ProductAlreadyExisits(payload))
+            {
+                return null;
+            }
+
+            _db.Add(product);
+            _db.SaveChanges();
+            return product;
+        }
+        public ProductItem? Get(int id)
+        {
+            return _db.ProductItems.FirstOrDefault(p => p.Id.Equals(id));
+        }
+
+        public List<ProductItem> GetAll()
+        {
+            return _db.ProductItems.ToList();
+        }
+
+        public List<ProductItem> GetAll(string category)
+        {
+            return _db.ProductItems.Where(p => p.Category.Equals(category, StringComparison.OrdinalIgnoreCase)).ToList();
+        }
+
+        public ProductItem? Update(int id, ProductPayload updatePayload)
+        {
+            if (ProductAlreadyExisits(updatePayload))
+            {
+                return null;
+            }
+
+            var product = Get(id);
+
+            if (!string.IsNullOrWhiteSpace(updatePayload.Name)) { product.Name = updatePayload.Name; }
+            if (!(updatePayload.Price < 0)) { product.Price = updatePayload.Price; }
+            if (!string.IsNullOrWhiteSpace(updatePayload.Category)) { product.Category = updatePayload.Category; }
+
+            _db.SaveChanges();
+
+            return product;
+
+        }
+        public ProductItem? Delete(int id)
+        {
+            var product = Get(id);
+            if (product == null) { return null; }
+
+            _db.ProductItems.Remove(product);
+            _db.SaveChanges();
+            return product;
```

Comment:
This doesnn't seem right. Returning an item after deleting it means you are sending stale data to the caller

Reply:
Ok connected to the comment above, is it more clear to return a bool then?
> It seems a bit odd to check if a product is null **after** deleting it. Unless I am misunderstanding your intent

I was thinking that it would be good to see what was deleted so therefore i return the stale data.

Reply:
I understand. But seeing what was deleted seems like an internal implementation detail which shouldn't be exposed to your public methods. In the real world, you'd log the deletion to an 'Audit' table and check it there